src/os/bluestore/BitAllocator.h

```diff
@@ -19,6 +18,18 @@
 #include "include/intarith.h"
 #include "os/bluestore/bluestore_types.h"
 
+#define alloc_assert assert
+
+#ifdef BIT_ALLOCATOR_DEBUG
+#define alloc_dbg_assert(x) assert(x)
+#else
+#define alloc_dbg_assert(x) noop(x)
+#endif
+
+static inline void noop(bool var)
+{
+}
```

Comment:
This might not get optimized out.  Better to do a simple

 #define alloc_dbg_assert(x)